### PR TITLE
[WIP] Use Koji to set a release label if one is absent

### PR DESCRIPTION
**Pull request by twaugh**
The bump_release plugin is not used in production anywhere right now so it is safe to re-use it for this purpose. The parameters it takes are now different.

### CODE REVIEW

File: atomic_reactor/plugins/pre_bump_release.py [hunk 1]
```diff
@@ -8,315 +8,63 @@
...
-
-        if branch_sha != self.git_ref:
-            self.log.error("Branch '%s' is at commit %s (expected %s)",
-                           branch, branch_sha, self.git_ref)
-            raise RuntimeError("Not at expected commit")
-
-        self.log.info("Branch '%s' is at expected commit (%s)",
-                      branch, self.git_ref)
+        self.target = target
+        self.xmlrpc = koji.ClientSession(hub)
```
Comment (vrutkovs) [inline atomic_reactor/plugins/pre_bump_release.py]: We probably should reuse ```create_koji_session``` from ```atomic_reactor.koji_util``` here

Comment (twaugh) [inline atomic_reactor/plugins/pre_bump_release.py]: We don't need to login for this. As written, `create_koji_session()` will try to use Kerberos if no credentials are supplied.

Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: It'll only attempt login if you provide a second parameter: https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/koji_util.py#L77

File: atomic_reactor/plugins/pre_bump_release.py [hunk 2]
```diff
@@ -8,315 +8,64 @@
...
+        latest = self.xmlrpc.getLatestBuilds(self.target, package=component)
+        try:
+            next_release = self.xmlrpc.getNextRelease(latest[0])
+        except IndexError:
+            next_release = 1
+
+        next_release = str(next_release)
+        for release_label in release_labels:
+            self.log.info("setting %s=%s", release_label, next_release)
+            dockerfile_labels[release_label] = next_release
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: Is this meant to actually write back to the Dockerfile? If so, I think you need an additional step: `parser.labels = dockerfile_labels`. https://github.com/DBuildService/dockerfile-parse/blob/master/dockerfile_parse/parser.py#L332

**Comment by lcarva** [comment]
We probably need changes in koji-containerbuild to properly perform n-v-r check. Maybe if release is not provided, bypass it completely?

File: atomic_reactor/plugins/pre_bump_release.py (hunk 2 above)
Comment (twaugh) [inline atomic_reactor/plugins/pre_bump_release.py]: `dockerfile_labels` is a `Labels` instance at this point. I should make that clearer.

**Comment by twaugh** [comment]
Yes, koji-containerbuild, osbs-client, and the assert_labels plugin -- and possibly other places -- will need to be able to handle the absence of a Release label.

File: atomic_reactor/plugins/pre_bump_release.py (hunk 2 above)
Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: Ah sweet! Didn't realize that `Label` class actually invokes the `parser.labels` property setter. A comment might help the next guy ;)

File: atomic_reactor/plugins/pre_bump_release.py (hunk 1 above)
Comment (twaugh) [inline atomic_reactor/plugins/pre_bump_release.py]: Aha.

**Comment by mmilata** [comment]
LGTM; -500 lines total, not bad;)

**Comment by lcarva** [comment]
LGTM

File: atomic_reactor/plugins/pre_bump_release.py [hunk 3]
```diff
@@ -8,315 +8,66 @@
...
-        :param author_name: str, name to use for git commits
-        :param author_email: str, email address for git commits
-        :param committer_name: str, name to use for git commits (else author's)
-        :param committer_email: str, email address for git commits (else
-                                     author's)
-        :param commit_message: str, git commit message
-        :param push_url: str, URL for push
+        :param target: string, koji target to use as a source
+        :param hub: string, koji hub (xmlrpc)
+        :param proxy: string, proxy user
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: Missing `proxy` parameter?

File: atomic_reactor/plugins/pre_bump_release.py [hunk 4]
```diff
@@ -8,315 +8,66 @@
...
+        component_label = get_preferred_label_key(dockerfile_labels,
+                                                  'com.redhat.component')
+        try:
+            component = dockerfile_labels[component_label]
+        except KeyError:
+            raise RuntimeError("missing label: {}".format(component_label))
 
-            # We checked out the right branch
-            assert repo.git(['rev-parse', 'HEAD']) == branch_sha
+        latest = self.xmlrpc.getLatestBuilds(self.target, package=component)
```
Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: `getLatestBuilds` uses `tag`, not `target`: https://pagure.io/koji/blob/master/f/hub/kojihub.py#_9111

Comment (lcarva) [inline atomic_reactor/plugins/pre_bump_release.py]: See https://github.com/projectatomic/atomic-reactor/pull/516 for fix.
Testing was successful.

File: atomic_reactor/plugins/pre_bump_release.py (hunk 3 above)
Comment (twaugh) [inline atomic_reactor/plugins/pre_bump_release.py]: Wrong comment, will fix in a separate PR.

**Pull request merged**